Changelog.md

```diff
@@ -9,6 +9,7 @@ Language Features:
 Compiler Features:
  * Command Line Interface: Do not perform IR optimization when only unoptimized IR is requested.
  * Commandline Interface: Add ``--transient-storage-layout`` output.
+ * Commandline Interface: Remove ``--yul`` flag.
```

Comment:
Is there are reason we're doing this now - it is *technically* a [breaking change](https://github.com/ethereum/solidity/blob/develop/docs/050-breaking-changes.rst?plain=1#L158-L159).

Reply:
As per discussion with @ekpyron it is a breaking change mainly on CLI level, not on language level b/c `--yul` isn't really usable anyways. Doing this to simplify yul dialect handling, prospectively getting rid of types, and finally make the whole numerical id integration easier.